### PR TITLE
mkvtoolnix: use wrapQtApp

### PR DESCRIPTION
**Pull request by vbgl**
See #65399

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/mkvtoolnix/default.nix [hunk 1]
```diff
@@ -4,6 +4,7 @@
 , withGUI ? true
   , qtbase ? null
   , qtmultimedia ? null
+  , wrapQtAppsHook ? null
```
Comment (jonringer) [inline pkgs/applications/video/mkvtoolnix/default.nix]: even though it's supposed to be denote it's optional, weird to see extra whitespace.

Personal preference. Don't feel strongly about it.
```suggestion
, wrapQtAppsHook ? null
```

File: pkgs/applications/video/mkvtoolnix/default.nix [hunk 2]
```diff
@@ -4,6 +4,7 @@
...
 }:
 
 assert withGUI -> qtbase != null && qtmultimedia != null;
```
Comment (jonringer) [inline pkgs/applications/video/mkvtoolnix/default.nix]:
```suggestion
assert withGUI -> qtbase != null && qtmultimedia != null && wrapQtAppsHook != null;
```

File: pkgs/applications/video/mkvtoolnix/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/applications/video/mkvtoolnix/default.nix]: I’ve just followed the style of the two lines above.

Comment (jonringer) [reply pkgs/applications/video/mkvtoolnix/default.nix]: either way, I don't have much of a preference. Just a slight preference :)

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
binaries seem to all work
leaf package

**Comment by teto** [comment]
@GrahamcOfBorg build mkvtoolnix

**Review by worldofpeace — APPROVED** [review]
I've reproduce the issue on master, upon building the package I ensured the correct binary was wrapped.

Executes without an error at runtime :heavy_check_mark:

**Pull request merged**